---
title: "RNAseq Analysis CyPPA Melody"
author: "Alejandro"
date: "2023-01-26"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown for Analyses of Yuequ Zhang's study of whole transcriptome of Neural Progenitor Cells treated with DMSO, and SK channel activators CyPPA and EBIO for different days.

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document.

Please install the relevant libraries before running the below chunks of text and make sure that the working directory is the directory containing the markdown file.

# Importing data

## Loading libraries and data "counts_wo_duplicates"

```{r message = FALSE, echo=FALSE}
#Libraries, make sure you have them all installed, the following lines are for opening them once they are installed
library(dplyr)
library(edgeR)
library(readxl)
library(EnhancedVolcano)
library(DESeq2)
library(dplyr)
library(readr)
library(ggplot2)
library(pheatmap)
library(edgeR)

counts_wo_duplicates <- read.csv("counts_wo_duplicates.csv", row.names=1, sep=";")

#These are raw counts, we need to filter out the lowly-expressed genes, as they are statistical noise. To do this, we will convert the raw counts to CPMs and will filter out any gene that has less than 0.5 cpms in 3 samples (our biological group size)

# Obtain CPMs
myCPM <- cpm(counts_wo_duplicates)
# Have a look at the output
head(myCPM)

col1sum <- sum(counts_wo_duplicates[,1])/1000000
counts_wo_duplicates[1,1]/col1sum

# Which values in myCPM are greater than 0.5?
thresh <- myCPM > 0.5
# This produces a logical matrix with TRUEs and FALSEs
head(thresh)
# Summary of how many TRUEs there are in each row
rowSums(head(thresh))
table(rowSums(thresh))

# we would like to keep genes that have at least 3 TRUES in each row of thresh
keep <- rowSums(thresh) >= 3
summary(keep)

# Subset the rows of countdata to keep the more highly expressed genes
counts.keep <- counts_wo_duplicates[keep,]
dim(counts_wo_duplicates)

dim(counts.keep)

  #After this step, you can see that from 60720 genes, the number of genes goes to 20256 in our 27 samples

#This filtering was based on https://sbc.shef.ac.uk/workshops/2018-07-10-rna-seq/rna-seq-preprocessing.nb.html#overview 


#Importing metadata
#Load metadata
library(readr)
metadata <- read_delim("metadata.csv", delim = ";", 
    escape_double = FALSE, trim_ws = TRUE)
```

## DESEQ2 Pipeline

```{r message=FALSE, warning=FALSE}
#make the datasetobject
library(DESeq2)
dds <- DESeqDataSetFromMatrix(countData = counts.keep, 
                              colData = metadata, 
                              design=~Condition + Replicate + Condition:Replicate) #I included the interaction between Condition and replicate because in the pca plots (See below), there was a huge difference between replicates.

#run the DESeq pipeline
dds <- DESeq(dds)
#PCAplot
rld <- rlog(dds)

#DMSO SAMPLES PCA PLOTS
rld.sub<-rld[,rld$Condition %in% c("DMSO_D0", "DMSO_D7", "DMSO_D14")]
plotPCA(rld.sub, intgroup= c("Condition")) #DMSO samples 

rld.sub<-rld[,rld$Condition %in% c("DMSO_D0", "DMSO_D7", "DMSO_D14")]
plotPCA(rld.sub, intgroup= c("Replicate")) #DMSO cluster by replicate 
#this pca plot shows that replicate or collection 1,2, and 3 are different among each other. this is why replicate should be taken into account into the design of the modeling for DE analysis

#EBIO SAMPLES PCA PLOTS
rld.sub<-rld[,rld$Condition %in% c("EBIO_D0", "EBIO_D7", "EBIO_D14")]
plotPCA(rld.sub, intgroup= c("Condition")) #subset of EBIO samples 

rld.sub<-rld[,rld$Condition %in% c("EBIO_D0", "EBIO_D7", "EBIO_D14")]
plotPCA(rld.sub, intgroup= c("Replicate")) #subset of EBIO samples cluster by Replicate number

#CyPPA samples
rld.sub<-rld[,rld$Condition %in% c("CyPPA_D0", "CyPPA_D7", "CyPPA_D14")]
plotPCA(rld.sub, intgroup= c("Condition")) #subset of CyPPA samples 

rld.sub<-rld[,rld$Condition %in% c("CyPPA_D0", "CyPPA_D7", "CyPPA_D14")]
plotPCA(rld.sub, intgroup= c("Replicate")) #subset of EBIO samples cluster by Replicate number

#Overview of all samples in supplementary figure
plotPCA(rld, intgroup = c("Condition"))
```

## Exporting normalized counts

Normalized counts are useful for graphic representation of read abundance per gene per sample. They are normalized by DESEQ. After discussing this with various experts in transcriptomics, we have concluded that **this is the best normalized gene count that we can use for further data exploration.**

In this study, these counts are normalized taking into account the interaction between replicate:condition

```{r message=FALSE, warning=FALSE}
NormalizedCounts<-counts(dds, normalized=T)
write.table(NormalizedCounts,file="normalizedCountsMelody.txt", sep=" ")
#These normalized counts can be plotted on prism for genes of interest
```

## Comparisons

```{r}
#EBIO
res_EBIOvsDMSO_D0 <- results(dds, contrast=c("Condition","EBIO_D0", "DMSO_D0"))
head(res_EBIOvsDMSO_D0)
res_EBIOvsDMSO_D0<- res_EBIOvsDMSO_D0[order(res_EBIOvsDMSO_D0$padj),]
head(res_EBIOvsDMSO_D0)
write.table(res_EBIOvsDMSO_D0, file="EBIO_D0 vs DMSO_D0.csv")
```

```{r}
#CyPPA
res_CYPPAvsDMSO_D0 <- results(dds, contrast=c("Condition","CyPPA_D0", "DMSO_D0"))
head(res_CYPPAvsDMSO_D0)
res_CYPPAvsDMSO_D0<- res_CYPPAvsDMSO_D0[order(res_CYPPAvsDMSO_D0$padj),]
head(res_CYPPAvsDMSO_D0)
write.table(res_CYPPAvsDMSO_D0, file="CyPPA vs DMSO_D0.csv")
```

## Creating Volcano Plot of comparisons of interest

```{r message=FALSE, warning=FALSE}
#EBIO VS DMSO D0 both
#Enhancedvolcanoplot for EBIO with labels as symbols.
TOP_EBIOVSDMSO_D0<-c(rownames(res_EBIOvsDMSO_D0))
top25_4_EBIOVSDMSO_D0<-TOP_EBIOVSDMSO_D0[1:25]

#First volcano plot with raw pvalues and logfold changes
EnhancedVolcano(res_EBIOvsDMSO_D0, 
                lab=rownames(res_EBIOvsDMSO_D0),
                selectLab = top25_4_EBIOVSDMSO_D0,
                x='log2FoldChange',
                y='pvalue',
                title= "EBIO vs DMSO 30 DAY TREATMENT (pval)",
                xlim=c(-10,10),
                ylim = c(0,5),
                pCutoff = 0.05,
                FCcutoff = 1.5,)


#Second volcano plot with raw pvalues and logfold changes
EnhancedVolcano(res_EBIOvsDMSO_D0, 
                lab=rownames(res_EBIOvsDMSO_D0),
                selectLab = top25_4_EBIOVSDMSO_D0,
                x='log2FoldChange',
                y='padj',
                title= "EBIO vs DMSO 30 DAY TREATMENT (padj)",
                xlim=c(-4,3),
                ylim = c(0,2),
                pCutoff = 0.05,
                FCcutoff = 1.5,)

#What I usually do is that I run the code to generate the volcano plot in a separate script to output it as a pdf with the graphical settings I want

#another option is to remove the first line of this code chunk, then the plot will appear in the plots tab
```

```{r message=FALSE, warning=FALSE}
#CYppa VS DMSO D0 both

#Enhancedvolcanoplot for EBIO with labels as symbols.
res_CYPPAvsDMSO_D0<- res_CYPPAvsDMSO_D0[order(res_CYPPAvsDMSO_D0$pvalue),]
TOP_CYPPAVSDMSO_D0<-c(rownames(res_CYPPAvsDMSO_D0))
top25_TOP_CYPPAVSDMSO_D0<-TOP_CYPPAVSDMSO_D0[1:25]


#CYPPA VS DMSO by raw pvalues
EnhancedVolcano(res_CYPPAvsDMSO_D0, 
                lab=rownames(res_CYPPAvsDMSO_D0),
                selectLab = top25_TOP_CYPPAVSDMSO_D0,
                x='log2FoldChange',
                y='pvalue',
                title= "CyPPA vs DMSO 30 DAY TREATMENT top dysregulated genes",
                xlim=c(-10,8),
                ylim = c(0,4),
                pCutoff = 0.05,
                FCcutoff = 1.5,) #this plot shows no significantly dysreguated genes


#CYPPA VS DMSO
EnhancedVolcano(res_CYPPAvsDMSO_D0, 
                lab=rownames(res_CYPPAvsDMSO_D0),
                selectLab = top25_TOP_CYPPAVSDMSO_D0,
                x='log2FoldChange',
                y='padj',
                title= "CyPPA vs DMSO 30 DAY TREATMENT top dysregulated genes",
                xlim=c(-4,3),
                ylim = c(0,3),
                pCutoff = 0.05,
                FCcutoff = 1.5,) #this plot shows no significantly dysreguated genes

#What I usually do is that I run the code to generate the volcano plot in a separate script to output it as a pdf with the graphical settings I want

#another option is to remove the first line of this code chunk, then the plot will appear in the plots tab
```
